storage/namespace.go

```diff
@@ -828,10 +882,16 @@ func (n *dbNamespace) initShards(needBootstrap bool) {
 
 func (n *dbNamespace) Close() error {
 	n.Lock()
+	if n.closed {
```

Comment:
Did we have a race condition before, or are you just protecting against duplicate calls?

Reply:
Just protecting against duplicate calls, seemed strange to be returning a potential error here but not actually returning one too.